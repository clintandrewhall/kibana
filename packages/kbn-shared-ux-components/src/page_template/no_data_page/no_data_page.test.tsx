/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React from 'react';
import { NoDataPage } from './no_data_page';
import { renderWithIntl } from '@kbn/test-jest-helpers';
import { NoDataCard } from '@kbn/shared-ux-card-no-data';
import { SharedUxServicesProvider, mockServicesFactory } from '@kbn/shared-ux-services';

describe('NoDataPage', () => {
  test('render', () => {
    const component = renderWithIntl(
      <SharedUxServicesProvider {...mockServicesFactory()}>
        <NoDataPage
          solution="Analytics"
          action={{
            elasticAgent: {},
          }}
          logo={'logoKibana'}
          docsLink="test"
        />
      </SharedUxServicesProvider>
    );
    expect(component).toMatchSnapshot();
    expect(component.find('h1').html()).toContain('Welcome to Elastic Analytics!');
    expect(component.find(NoDataCard).length).toBe(1);
  });
});
