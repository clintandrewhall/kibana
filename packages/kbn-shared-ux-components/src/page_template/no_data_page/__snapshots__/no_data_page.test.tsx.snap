// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoDataPage render 1`] = `
<div
  class="kbnNoDataPageContents"
  data-test-subj="kbnNoDataPage"
>
  <div
    class="euiText euiText--medium"
  >
    <div
      class="euiTextAlign euiTextAlign--center"
    >
      <div
        aria-label="Analytics"
        class="euiAvatar euiAvatar--xl euiAvatar--user kbnSolutionAvatar kbnSolutionAvatar--xxl css-1adte2t-euiAvatar-xl-user-plain"
        role="img"
        title="Analytics"
      >
        <span
          class="euiAvatar__icon"
          data-euiicon-type="logoKibana"
        >
          Analytics
        </span>
      </div>
      <div
        class="euiSpacer euiSpacer--l css-p2o3x6-euiSpacer-l"
      />
      <h1>
        Welcome to Elastic Analytics!
      </h1>
      <span
        class="euiTextColor euiTextColor--subdued"
      >
        <p>
          Add your data to get started, or 
          <a
            class="euiLink css-102pf9n-euiLink-primary"
            href="test"
            rel="noopener noreferrer"
            target="_blank"
          >
            learn more
            <span
              class="css-q4vwwx-euiLink__externalIcon"
              data-euiicon-type="popout"
            >
              External link
            </span>
            <span
              class="euiScreenReaderOnly"
            >
              (opens in a new tab or window)
            </span>
          </a>
           about Analytics.
        </p>
      </span>
    </div>
  </div>
  <div
    class="euiSpacer euiSpacer--xxl css-1ryjopu-euiSpacer-xxl"
  />
  <div>
    <div
      class="euiPanel euiPanel--paddingLarge euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPanel--isClickable euiCard euiCard--centerAligned euiCard--isClickable css-1hu4pg0-EuiCard"
    >
      <div
        class="euiCard__top"
      >
        <div
          class="euiCard__image"
        >
          <figure
            class="euiImage euiImage--fullWidth"
          >
            <img
              alt=""
              class="euiImage__img"
              src="test-file-stub"
              style="width:max(100%, 360px);height:240px;object-fit:cover;background:aliceblue"
            />
          </figure>
        </div>
      </div>
      <div
        class="euiCard__content"
      >
        <span
          class="euiTitle euiCard__title css-1mr6cet-euiTitle-s"
          id="generated-idTitle"
        >
          <a
            aria-describedby="generated-idDescription"
            class="euiCard__titleAnchor"
            href="/app/integrations/browse"
            rel="noreferrer"
          >
            <span
              class="euiScreenReaderOnly"
            >
              Add Elastic Agent
            </span>
          </a>
        </span>
        <div
          class="euiText euiText--small euiCard__description"
          id="generated-idDescription"
        >
          <p>
            Use Elastic Agent for a simple, unified way to collect data from your machines.
          </p>
        </div>
      </div>
      <div
        class="euiCard__footer"
      >
        <button
          class="euiButton euiButton--primary euiButton--fill"
          type="button"
        >
          <span
            class="euiButtonContent euiButton__content"
          >
            <span
              class="euiButton__text"
            >
              Add Elastic Agent
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
